scripts/components/Main.js

```diff
@@ -138,6 +138,18 @@ export default class Main extends React.Component {
         }
       }
     }
+
+    //Makes sure the user is warned before closing the window
+    window.addEventListener('beforeunload', function (e) {
+      if(e.target.body.firstChild.classList.contains("h5p-threeimage-editor")
+        || this.state.scoreCard.numQuestionsInTour === 0
+        || this.state.scoreCard.totalQuestionsCompleted === 0)
+      {
+        return;
+      }
+      e.preventDefault();
+      e.returnValue = '';
+    }.bind(this));
```

Comment:
Use arrow functions to keep the parent `this`

```suggestion
    window.addEventListener('beforeunload', (e) => {
      if(e.target.body.firstChild.classList.contains("h5p-threeimage-editor")
        || this.state.scoreCard.numQuestionsInTour === 0
        || this.state.scoreCard.totalQuestionsCompleted === 0)
      {
        return;
      }
      e.preventDefault();
      e.returnValue = '';
    });
```